realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -282,6 +284,47 @@ public void refresh() {
     }
 
     /**
+     * Blocks the current thread until new changes to the Realm are available or {@link #stopWaitForChange()}
+     * is called from another thread. Once stopWaitForChange is called, all future calls to this will
+     * return false immediately.
+     *
+     * @return {@code true} If the Realm was updated to the latest version, {@code false} if it was
+     * cancelled by calling stopWaitForChange.
+     * @throws IllegalStateException if calling this from within a transaction or from a Looper thread.
+     */
+    public boolean waitForChange() {
+        checkIfValid();
+        if (isInTransaction()) {
+            throw new IllegalStateException("Cannot wait for changes inside of a transaction.");
+        }
+        if (Looper.myLooper() != null) {
+            throw new IllegalStateException("Cannot wait for changes inside a Looper thread. Use RealmChangeListeners instead.");
+        }
+        boolean hasChanged = sharedGroupManager.getSharedGroup().waitForChange();
+        if (hasChanged) {
+            // Since this Realm instance has been waiting for change, advance realm & refresh realm.
+            sharedGroupManager.advanceRead();
+            handlerController.refreshSynchronousTableViews();
+        }
+        return hasChanged;
+    }
+
+    /**
+     * Makes any current {@link #waitForChange()} return {@code false} immediately. Once this is called,
+     * all future calls to waitForChange will immediately return {@code false}.
+     * <p>
+     * This method is thread-safe and should _only_ be called from another thread than the one that
+     * called waitForChange.
+     */
+    public void stopWaitForChange() {
+        // Check if the Realm instance has been closed
+        if (sharedGroupManager == null || !sharedGroupManager.isOpen()) {
+            throw new IllegalStateException(BaseRealm.CLOSED_REALM_MESSAGE);
+        }
+        sharedGroupManager.getSharedGroup().stopWaitForChange();
```

Comment:
We could run this method inside a synchrnoized block on `RealmCache`, two options:
1. just
```java
public void stopWaitForChange() {
    sychronized (RealmCache.class) {
        //....
    }
}
```

2. Create a method `static synchronized void invokeWithLock(Callback0 callback)` in `RealmCache` which is similar to `RealmCache.invokeWithGlobalRefCount`

I prefer the 2nd option since it leave the synchronized keyword inside `RealmCache` which is more readable.